src/docs/arch-block.dox

```diff
@@ -1,9 +1,195 @@
 /*! @arch_page arch-block Block Manager
 
-The Block Manager manages the reading and writing of disk blocks
-in WiredTiger.  It does compression and encryption when these
-are configured.
+The WiredTiger block manager subsystem manages the reading and writing of data
+from the disk. It is designed to facilitate high performance, economic use of
+disk space and customizability.
+
+@section block What is a block?
+
+A block is a chunk of data that is stored on the disk and operated on as a
+single unit. Each WiredTiger data file (any file in the home directory with the
+\c .wt suffix) is made up of these blocks. Each block consists of a page header,
+a block header and contains a single page of the btree from which it was
+generated. Blocks can only be written once and a new block is created upon
+reconciliation of a dirty page. The size of a block is a multiple of the
+allocation size which is set during creation of the associated WiredTiger data
+file see: WT_SESSION::create.
+
+Once a block is written an address cookie is returned. This address cookie is
```

Comment:
_This address cookie is stored as the \c addr on the associated page ref._

^ This kind of comes out of nowhere. Could we reference a source file, perhaps?

Reply:
Yeah it does come out of nowhere but I feel like it is helpful, as address cookie to addr wouldn't be obvious to me without knowing that already. I'll elaborate a little bit

Reply:
240571f